compiler/deSugar/DsBinds.hs

```diff
@@ -1140,7 +1140,9 @@ dsHsWrapper (WpCast co)       = ASSERT(coercionRole co == Representational)
                                 return $ \e -> mkCastDs e co
 dsHsWrapper (WpEvApp tm)      = do { core_tm <- dsEvTerm tm
                                    ; return (\e -> App e core_tm) }
-
+dsHsWrapper (WpMultCoercion co) = do { when (not (isReflCo co)) $
```

Comment:
As as side remark, there may be coercions which are functionally reflexivity but where `isReflCo` returns false, I believe. Typically, the transitivity rule applies to two reflexivity. We may want, at some point to be resilient to that.

I'm just raising this point for the sake of awareness.

Reply:
Yes -- use `isReflexiveCo` instead.